### DISCUSSION TITLE
Issue with Agnoster git branch icon

### DISCUSSION THREAD

**Discussion by govindkailas** [ohmybash/oh-my-bash#473] [Q&A] [answered]
I am using the `Agnoster` theme, powerline10 fonts are installed. I see a square symbol instead of the git branch symbol. 
![image](https://github.com/ohmybash/oh-my-bash/assets/6698224/fbf6b79f-6b07-4c64-baff-81a0a231ea59)

```
 echo "$PS1" | cat -v
\[^[]633;A^G\]\[^[[0m\]\[^[[0;40m\] gkailas@\h \[^[[30;44m\]M-nM-^BM-0\[^[[0;44;30m\] \w \[^[[34;42m\]M-nM-^BM-0\[^[[0;42;30m\] M-nM-^BM-  main \[^[[0;32m\]M-nM-^BM-0 \[^[[0m\]\[^[]633;B^G\]
```
What could be going wrong?

**Comment by akinomyoga** [comment]
You need to configure a powerline font supporting U+E0A0 in the terminal you use.

**Reply by govindkailas** [reply]
I'm on `WSL` , terminal from `vscode`. Any suggestion on which font should be used/supported?

**Reply by akinomyoga** [reply]
In the first place, did you configure any font in the VS Code terminal? U+E0A0 is a part of the basic set of powerline fonts, so it should be supported by presumably all the powerline fonts. I suspect that you just installed powerline fonts in the system but did not specify one for the VS Code terminal.

**Reply by akinomyoga** [reply]
Also, please note that the VS Code Terminal supports the triangle symbols by default, so the triangle symbols are displayed even when no powerline font is specified. The correctly shown triangle symbols do not mean that a powerline font is enabled in the VS Code Terminal.

**Reply by govindkailas** [reply]
I followed this https://code.visualstudio.com/docs/terminal/appearance 
As soon as I add other fonts it shows this small pop-up window in the corner saying Terminal support only `monospace`
![image](https://github.com/ohmybash/oh-my-bash/assets/6698224/93d6ecf3-7c2c-4f62-b20d-6adc7ce1545f)

**Reply by akinomyoga** [reply]
That popup also appears while typing the font name when the font name is incomplete. If the branch symbol (U+E0A0) is correctly shown after finishing inputting the font name, you don't need to care about the error message. If it doesn't still show the branch symbol, the font name might be wrong. Could you check it again? Note that font names containing whitespaces need to be quoted within `'...'`. Also, you may specify `monospace` as a fallback. For example, `'Your Powerline Font Name', monospace`.

**Comment by akinomyoga** [comment]
May I ask how did you "install" the powerline fonts?

What kind of powerline fonts do you install? There are many powerline fonts including [`powerline/fonts`](https://github.com/powerline/fonts) and others. Are the powerline fonts you installed the ones from `powerline/fonts`? If this is the case, the instruction on README of `powerline/fonts` is the way to install the fonts in the remote WSL system, so it doesn't make the fonts available to VS Code which is a client in the Windows subsystem.

You need to install the fonts in the Windows subsystem, i.e., you need to download the TTF files corresponding to the font you would like to use and install these TTF files through the Windows interface.

**Reply by govindkailas** [reply, marked as answer]
I installed it by cloning the repo and running `install.ps1` . I figured out that the fonts were getting installed with a different name, opened MS Word and noticed the name was `Fira Mono for Powerline`. Updated it as the Terminal Font family and it worked :) 
![image](https://github.com/ohmybash/oh-my-bash/assets/6698224/88a09713-8719-40ef-a816-8531b20ab105)

**Reply by akinomyoga** [reply]
Great!
